mars_troughs/trough.py

```diff
@@ -77,7 +78,17 @@ def __init__(
         self.re2_data_spline = RBS(self.lags, self.ins_times, self.retreats ** 2)
 
         # Calculate the model of lag per time 
-        self.lag_model_t = self.get_lag_model_t(self.ins_times)
+        num=self.lag_model_number
+        p=self.lag_params
+        
+        if num == 0: 
+            constant = p[0]
+            constant_lag=ConstantLag(constant)
```

Comment:
I think these code blocks need changing. The idea is to keep the lag model around. So instead of instantiating the lag model (e.g. `ConstantLag`) and then computing it once, we want to attach the lag model itself to the trough and then update it farther down.

E.g.:
```python
def __init__(self, ...):
    ...
    if self.lag_model_number == 0:
        self.lag_model = ConstantLag(*lag_params)
    elif self.lag_model_number == 1:
        self.lag_model = LinearLag(*lag_params)

    self.lag_model_t = self.lag_model.get_lag_at_t(ins_times)
    self.retreat_model_t = self.get_retreat_model_t(self.lag_model_t, ins_time)
    ...

def set_model(self, acc_params, lag_params, errorbar):
    ...
    self.lag_model.update_params(*lag_params)  # may need to implement this method
    ...
```

Does this make sense? We can get on a call sometime to talk about this.

Reply:
Done